### PR TITLE
Support dumping out rendered cloudformation templates for each stack

### PR DESCRIPTION
**Pull request by mhahn**
Output looks like:

```
$ stacker build -r us-west-2 conf/lunohq/development.env conf/lunohq/config.yaml --dump dump
[2016-05-17T14:05:33] Dumping "Create/Update stacks"...
[2016-05-17T14:05:33] Writing stack "dev-lunohq-vpc" -> dump/vpc-3ea2d331.json
[2016-05-17T14:05:33] Writing stack "dev-lunohq-empireDB" -> dump/empireDB-2bc493a4.json
[2016-05-17T14:05:33] Writing stack "dev-lunohq-empireMinion" -> dump/empireMinion-6b3bcb53.json
[2016-05-17T14:05:33] Writing stack "dev-lunohq-empireController" -> dump/empireController-92678c3c.json
...
````

### CODE REVIEW

File: stacker/actions/build.py [hunk 1]
```diff
@@ -290,22 +290,24 @@ def pre_run(self, outline=False, *args, **kwargs):
...
         """Kicks off the build/update of the stacks in the stack_definitions.
 
         This is the main entry point for the Builder.
 
         """
         plan = self._generate_plan(tail=tail)
-        if not outline:
+        if outline:
+            plan.outline()
+        elif dump:
```
Comment (phobologic) [inline stacker/actions/build.py]: Doesn't seem like these need to be mutually exclusive, right?  The command doesn't give any feedback like that, so might as well allow both.  Maybe ```
if not outline and not dump:
    ... actual build ...
else:
    if outline:
        ... outline ...
    if dump:
        ... dump ...
```

?

File: stacker/plan.py [hunk 2]
```diff
@@ -282,6 +285,24 @@ def outline(self, level=logging.INFO, message=''):
         if message:
             logger.log(level, message)
 
+    def dump(self, output):
```
Comment (phobologic) [inline stacker/plan.py]: maybe change output to directory?

Comment (phobologic) [inline stacker/plan.py]: Also, can you include `os.path.expanduser`?

**Comment by phobologic** [comment]
Awesome 👍 Did you get a chance to look at the local parameters 'issue'?

File: stacker/actions/build.py (hunk 1 above)
Comment (mhahn) [inline stacker/actions/build.py]: ah yea, we can do that.

**Comment by ejholmes** [comment]
Sweet 👍 this will be really nice to have.

I was thinking it would be cool if stacker just compiled all of the blueprints down to a single "metastack" template, which could then be executed by stacker itself. If we exposed a subcommand to execute a metastack template, then this could open up other use cases for stacker (people that want the "stack glue" aspect of stacker, but not troposphere/python), and you could just dump the metastack to stdout (e.g. `stacker compile | stacker execute`).

Anyways, definitely not something we should do in this PR, just something I was thinking about.

**Comment by phobologic** [comment]
Honestly for the non-troposphere/python users, I think we should provide a concept of 'drivers'.  You don't need to use a Blueprint in that case, we could have a 'json' driver or something that just reads a regular old CF template and uses that.

**Pull request merged**